### PR TITLE
Changed all the html into go using go-elem

### PR DESCRIPTION
**Pull request by amha-mersha**
Created templates package in ./hscontrol/templates. Moved the registerWebAPITemplate into the templates package as a function to be called.

Replaced the apple and windows html files with go-elem.

- [X] have read the [CONTRIBUTING.md](./CONTRIBUTING.md) file
- [ ] raised a GitHub issue or discussed it on the projects chat beforehand
- [ ] added unit tests
- [ ] added integration tests
- [ ] updated documentation if needed
- [ ] updated CHANGELOG.md

### CODE REVIEW

**Review by kradalby — APPROVED** [review]

**Comment by kradalby** [comment]
@nblock if you have some time, can you do a quick skim of the content for windows and apple?

**Comment by kradalby** [comment]
@amha-mersha can you delete the HTML templates so we are sure noone will update them in error?

**Comment by nblock** [comment]
> @nblock if you have some time, can you do a quick skim of the content for windows and apple?

Sure, but I got this while building this commit:

```
error: hash mismatch in fixed-output derivation '/nix/store/xj7li14vnd7v6rnmlj5k7v2rqgm7hxw7-headscale-1e8fc3e-go-modules.drv':
         specified: sha256-+8dOxPG/Q+wuHgRwwWqdphHOuop0W9dVyClyQuh7aRc=
            got:    sha256-U5u4tRcCQ+aRIFzOOm0KWRX4BitWKwiHtq5m2DadNtA=
error: 1 dependencies of derivation '/nix/store/f2ds1pcql2haf3s2z3m68mfikcga7lix-headscale-1e8fc3e.drv' failed to build
```

File: hscontrol/templates/windows.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+func Windows(url string) *elem.Element {
+	return HtmlStructure(
+		elem.Title(nil,
+			elem.Text("headscale - Windows"),
+		),
+		elem.Body(attrs.Props{
+			attrs.Style : bodyStyle.ToInline(),
+		},
+			headerOne("headscale: Windows configuration"),
```
Comment (nblock) [inline hscontrol/templates/windows.go:18]: space missing:

image

File: hscontrol/templates/apple.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+		elem.Title(nil,
+			elem.Text("headscale - Apple")),
+		elem.Body(attrs.Props{
+			attrs.Style: bodyStyle.ToInline(),
+		},
+			headerOne("headscale: iOS configuration"),
+			headerTwo("GUI"),
+			elem.Ol(nil,
+				elem.Li(nil,
+					elem.Text("Install the official Tailscale iOS client from the"),
```
Comment (nblock) [inline hscontrol/templates/apple.go:21]: The space between `from the` and the appstore link is missing:

image

File: hscontrol/templates/apple.go [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+					elem.A(attrs.Props{attrs.Href: "https://apps.apple.com/app/tailscale/id1470499037"},
+						elem.Text("App store"),
+					),
+				),
+				elem.Li(nil,
+					elem.Text("Open Tailscale and make sure you are"),
+					elem.I(nil, elem.Text("not")),
```
Comment (nblock) [inline hscontrol/templates/apple.go:28]: When rendered, there is no space between those elements:


image

File: hscontrol/templates/apple.go [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+							elem.Text(`If the iOS device was previously logged into Tailscale, switch the "Reset Keychain" toggle to "on"`),
+						),
+					),
+				),
+				elem.Li(nil,
+					elem.Text(fmt.Sprintf(`Enter "%s" under "Alternate Coordination Server URL`,url)),
+				),
+				elem.Li(nil,
+					elem.Text("Restart the app by closing it from the iOS app switcher, open the app and select the regular sign in option"),
+					elem.I(nil, elem.Text("(non-SSO)")),
```
Comment (nblock) [inline hscontrol/templates/apple.go:50]: missing space: 

image

File: hscontrol/templates/apple.go [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+				elem.Li(nil,
+					elem.Text("Find Tailscale and select it"),
+					elem.Ul(nil,
+						elem.Li(nil,
+							elem.Text(`If the iOS device was previously logged into Tailscale, switch the "Reset Keychain" toggle to "on"`),
+						),
+					),
+				),
+				elem.Li(nil,
+					elem.Text(fmt.Sprintf(`Enter "%s" under "Alternate Coordination Server URL`,url)),
```
Comment (nblock) [inline hscontrol/templates/apple.go:46]: The closing `"` is missing after `Coordination Server URL`

**Review by nblock — CHANGES_REQUESTED** [review]
Some spaces are missing between element, probably due to minified output (left a few screenshots where spotted).

**Comment by amha-mersha** [comment]
@nblock I think it is now fixed

**Comment by nblock** [comment]
> @nblock I think it is now fixed

That was quick, thx! 

Found one additional location where the space is lost: 

image

**Comment by amha-mersha** [comment]
how about now @nblock . sorry to take your time for this things

**Review by nblock — APPROVED** [review]

**Comment by kradalby** [comment]
Linter and `TestAuthKeyLogoutAndRelogin` is expected, but can you have a look at updating the build flake sha?

**Comment by kradalby** [comment]
A rebase should also fix the failing integration tests.

**Comment by amha-mersha** [comment]
So should I first rebase and commit

**Comment by kradalby** [comment]
yep that sounds best

**Comment by amha-mersha** [comment]
I have rebased and pushed again

**Comment by kradalby** [comment]
Think the nix flake needs to be update again, haha, we are very close!

**Comment by amha-mersha** [comment]
do I update that and how

**Comment by kradalby** [comment]
In the failing build there is a new hash that needs to replace the old one one flake.nix, so a search replace should be good.
 Sent from Proton Mail for iOS 
On Thu, Oct 3, 2024 at 13:44, Amha Mersha ***@***.***&gt; wrote:   do I update that and how —

**Comment by kradalby** [comment]
I fixed it up and will merge it when the test runs :) Thanks @amha-mersha, if you feel like more challenges there is one linked in the issue for OIDC page that is trickier because it has more CSS.

**Comment by amha-mersha** [comment]
Thank you @kradalby for taking the time and guiding me. And for the other one, sure I look into it and get back to you

**Comment by amha-mersha** [comment]
Could you share me the issue, I couldn't find it.

**Comment by kradalby** [comment]
https://github.com/juanfont/headscale/issues/2115 @amha-mersha

**Pull request merged**